### PR TITLE
Adds some more colors to the output of certain setup cli commands

### LINKED ISSUE #29335: [Issue] Adds some more colors to the output of certain setup cli commands
This issue is automatically created based on existing pull request: magento/magento2#29298: Adds some more colors to the output of certain setup cli commands

---------
### Description (*)
While trying to figure out the sort order in which certain database migration scripts ran while executing `bin/magento setup:upgrade`, I found the output a bit hard to read.
So I've added a different color to the output of certain steps.

Before:
before

After:
after

This also influences some other `bin/magento setup:xxx` commands but it doesn't seem to degrade them in my opinion (but I haven't tested all of them).

### Fixed Issues (if relevant)
None that I could find

### Manual testing scenarios (*)
1. Execute the `bin/magento setup:upgrade` command and look if the colors used in the output make sense
2. Execute the other `bin/magento setup:xxx` commands and do the same

### Questions or comments
If you want different colors or if there are better idea's, let me know!
I've just re-used the "metadata" color which was already being used.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### PR DESCRIPTION
**Pull request by hostep**
### Description (*)
While trying to figure out the sort order in which certain database migration scripts ran while executing `bin/magento setup:upgrade`, I found the output a bit hard to read.
So I've added a different color to the output of certain steps.

Before:
before

After:
after

This also influences some other `bin/magento setup:xxx` commands but it doesn't seem to degrade them in my opinion (but I haven't tested all of them).

### Fixed Issues (if relevant)
None that I could find

### Manual testing scenarios (*)
1. Execute the `bin/magento setup:upgrade` command and look if the colors used in the output make sense
2. Execute all other console commands that use the ConsoleLoggerInterface class and see if the colors used in the output make sense

### Questions or comments
If you want different colors or if there are better idea's, let me know!
I've just re-used the "metadata" color which was already being used.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### Resolved issues:
1. [x] resolves magento/magento2#29335: Adds some more colors to the output of certain setup cli commands

### CODE REVIEW

**Comment by hostep** [comment]
@magento run Static Tests

**Comment by hostep** [comment]
@magento run Static Tests

**Comment by hostep** [comment]
@magento run all tests

**Comment by hostep** [comment]
Before I put more effort in this, can I know if this improvement will be blocked by the failing Semantic Version Checker? Or should an architect first review the introduction of a new method to the interface and give his/her approval?

**Comment by VladimirZaets** [comment]
@magento create issue

File: lib/internal/Magento/Framework/Setup/LoggerInterface.php
Comment (sidolov) [inline lib/internal/Magento/Framework/Setup/LoggerInterface.php]: I prefer to introduce such changes in backward compatible way - introduce the new interface and implementation for that.

**Review by sidolov — CHANGES_REQUESTED** [review]

File: lib/internal/Magento/Framework/Setup/LoggerInterface.php
Comment (hostep) [reply lib/internal/Magento/Framework/Setup/LoggerInterface.php]: Sure, my biggest problem with introducing new interfaces is: how do you name them? @sidolov: do you have a suggestion?

Comment (sidolov) [reply lib/internal/Magento/Framework/Setup/LoggerInterface.php]: `LogMetafataInlineInterface` with method `execute`?

Comment (hostep) [reply lib/internal/Magento/Framework/Setup/LoggerInterface.php]: So each time we add a new method to an interface, we have to create a new interface with the name of the new method prepended to its name? So next time we add a `logWhatever` method the new interface will be called `LogWhateverLogMetadataInlineInterface` ???

This is soo silly. Can't we just get approval from an architect to get this method added to the original interface? How many people will have written a new implementation of this interface you think? Not a lot I guess? So it has a low probability of causing backwards incompatible issues I guess and if somebody runs against it it should be very clear to them what the issue is and they should be able to fix it very quickly.

But that's just what I think, if some architect could get involved here to leave a comment, that would be appreciated!

**Comment by gabrieldagama** [comment]
The risk was set to`low` due to: the suggested changes shouldn't affect other areas.

File: lib/internal/Magento/Framework/Setup/LoggerInterface.php
Comment (sidolov) [reply lib/internal/Magento/Framework/Setup/LoggerInterface.php]: I got your point but still, we will have a small percent of developers who implemented the interface and their implementation will be broken after such changes. That's the reason to have strict rules - to support 3d party code and make it reliable when you are using the core code marked as @api.

Comment (hostep) [reply lib/internal/Magento/Framework/Setup/LoggerInterface.php]: Hi @sidolov 

I know the reasoning, but I very often see exceptions being allowed where no extra interfaces are required.
We already discussed this before in https://github.com/magento/magento2/pull/26748#issuecomment-593558031

Thanks

**Comment by engcom-Golf** [comment]
Hi @hostep, thank you for your contribution, I will take care of the required changes on the PR.

**Comment by engcom-Golf** [comment]
@magento run all tests

**Comment by engcom-Hotel** [comment]
@magento run all tests

**Comment by hostep** [comment]
Can we have any sort of progress update here? Thanks! 🙂

**Comment by korostii** [comment]
@engcom-Golf @VladimirZaets @sidolov Respectfully, is someone looking through the older and low priority PRs nowadays, at all? How is it possible that they can be stuck for half a year with no progress from Magento end, at all? 
Thanks in advance.

**Comment by hostep** [comment]
@korostii, it's not really a high prio issue to get this merged, but some kind of update would indeed be appreciated :)

Leaving PR's lying around, giving no updates is very discouraging to contributors. I've recently saw a discussion in the chat of MMUK21 where two very well known and experienced Magento contributors complained about the same thing and said it had no use anymore to send in PR's to Magento as they aren't getting processed in a timely manner. This is very sad to read, and I really hope the Magento maintainers will change this one day, preferably sooner than later. Adhering to priorities in PR's doesn't make a lot of sense because it means low prio PR's will never get processed. I complained about this before in https://github.com/magento/magento2/pull/28927#issuecomment-861650918 BTW but nothing seems to happen 😞

File: lib/internal/Magento/Framework/Setup/LoggerInterface.php
Comment (hostep) [reply lib/internal/Magento/Framework/Setup/LoggerInterface.php]: Let's resolve this because Viktor Kopin introduced a new `ConsoleLoggerInterface` class.

**Comment by hostep** [comment]
Rebased on latest `2.4-develop` branch, because there was a conflict with the `BackupRollback.php` file, so that needed fixing.

Unit tests will still be broken, still would like some help with fixing those if somebody's interested 🙂 

@magento run all tests

**Comment by hostep** [comment]
@magento run all tests

**Comment by hostep** [comment]
I think I've solved the hard-to-solve unit tests. Also rebased the whole thing on the latest `2.4-develop` branch.

Let's see how many new tests will fail after having left this PR open for soo long.

**Comment by engcom-Hotel** [comment]
@magento run all tests

**Comment by hostep** [comment]
Seems like no tests wanted to run, so let's try again

@magento run all tests

**Review by engcom-Hotel — APPROVED** [review]
Hello @hostep,

Thanks for the contribution!

Functional test failures seem flaky to me. And for Semantic version check failure, we will ask for approval.

Hence approving this PR.

Thanks

**Comment by engcom-Lima** [comment]
@magento run all tests

**Comment by engcom-Lima** [comment]
### :heavy_check_mark: QA Passed

**Preconditions:**
- Install fresh Magento **`2.4-develop`** and **`PHP 8.1`**

**Manual testing scenario:**
- Execute below commands:
     -  `bin/magento setup:upgrade`
     -  `bin/magento  setup:db-data:upgrade`
     -  `bin/magento setup:install --base-url=http://local.magento2.com` 

**Before:** :heavy_multiplication_x: 
Screenshot 2023-06-12 at 2 34 24 PM
Screenshot 2023-06-12 at 4 54 42 PM


**After:** :heavy_check_mark:  
Screenshot 2023-06-12 at 2 38 12 PM
Screenshot 2023-06-12 at 2 38 03 PM
Screenshot 2023-06-12 at 4 51 18 PM

Screenshot 2023-06-12 at 4 51 38 PM

Screenshot 2023-06-12 at 4 54 54 PM

No additional manual test cases is required as part of regression as this PR is related to the adding of colors to the CLI commands

Builds are failed. Hence, moving this PR to **Extended Testing.**

**Comment by engcom-Echo** [comment]
@magento run Functional Tests CE

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by engcom-Echo** [comment]
@magento run Functional Tests EE,Functional Tests B2B

**Comment by engcom-Echo** [comment]
Functional Tests EE and Functional Tests B2B failure are different on last two run on same commit.
Functional Tests B2B
Run1
image

Run2
image

Functional Tests EE
Run1
image


Run2
image

For  Semantic Version Checker  failure we have created internal [JIRA](https://jira.corp.adobe.com/browse/AC-8995) ticket. Hence moving to `Pending Approval`

**Comment by hostep** [comment]
I've merged `2.4-develop` in here to deal with merge conflicts caused by:
- changes introduced in https://github.com/magento/magento2/pull/38123
- needed to update amount of modules (from 22 to 23 or 23 to 24) in the Installer unit test, possibly by introduction of new modules to core Magento in the past few weeks, just guessing here

Resolved conflicts in files:
- setup/src/Magento/Setup/Model/Installer.php
- setup/src/Magento/Setup/Test/Unit/Model/InstallerTest.php

PR should be fully re-tested, both automatically as manually.


@ magento run all tests

**Comment by hostep** [comment]
@magento run all tests

**Comment by engcom-Charlie** [comment]
As per [this](https://github.com/magento/magento2/pull/29298#issuecomment-1870649012) comment, moving it to Ready for Testing again.

**Comment by engcom-Bravo** [comment]
@magento run all tests

**Comment by hostep** [comment]
Thanks! But your screenshots under Before & After got upload in the wrong order I think, some of those need to be moved from Before to After and some from After to Before.

But happy to see it works as expected.

**Comment by engcom-Charlie** [comment]
@magento run Functional Tests EE,Functional Tests B2B, WebAPI Tests

**Comment by engcom-Echo** [comment]
Moving it to `Pending Approval` as the internal [JIRA](https://jira.corp.adobe.com/browse/AC-8995) is yet to be approved for Semantic Version Checker failure

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by engcom-Echo** [comment]
We have received approval for internal [JIRA](https://jira.corp.adobe.com/browse/AC-8995) ticket.
Hence moving it to `Extended Testing`

**Comment by engcom-Echo** [comment]
@magento run Functional Tests EE ,Functional Tests B2B ,WebAPI Tests

**Comment by engcom-Echo** [comment]
@magento run Functional Tests B2B

**Comment by engcom-Echo** [comment]
Functional Tests B2B failures contain one(_**LYNX-180: Customer cancels an order from order history page.**_) common failure on last two run which is not related to PR changes.
Run1
Screenshot 2024-03-20 at 8 05 10 AM

Run2
Screenshot 2024-03-20 at 8 05 03 AM

**Pull request merged**
